src/thunderbots/software/ai/hl/stp/tactic/shadow_enemy_tactic.cpp

```diff
@@ -25,14 +27,17 @@ std::string ShadowEnemyTactic::getName() const
 void ShadowEnemyTactic::updateParams(const Evaluation::EnemyThreat &enemy_threat,
                                      const Field &field, const Team &friendly_team,
                                      const Team &enemy_team, double shadow_distance,
-                                     bool enemy_team_can_pass)
+                                     bool enemy_team_can_pass, const Ball &ball,
+                                     const double ball_steal_speed)
```

Comment:
Yes all of these need to be updated every time because some are parameters that need to reflect changed values. I think this should be cleaned up in the planned STP changes to make Plays / Tactics simpler.

Reply:
Even with that change tactics will still have an `updateParams` method so that they're testable. I don't think we change `enemy_team_can_pass`, `ball_steal_speed`, `shadow_distance` each time here? Correct me if I'm wrong but I think we can pass these into the constructor.

Reply:
`ball_steal_speed` and `shadow_distance` can be moved since we don't have a use case for needing to update them on the fly yet. `enemy_team_can_pass` needs to stay since it's a parameter than can change during runtime

Reply:
Correction, only `ball_steal_speed` moved to the constructor. The `KickoffEnemy` play uses the shadow_distance as a parameter